tests/cases/compiler/noImplicitAnySpreads1.ts

```diff
@@ -0,0 +1,19 @@
+// @strictNullChecks: false
+// @noImplicitAny: true
```

Comment:
I default all the tests I add to `@strict: true, @noEmit: true`. This combination of flags is important to showcase the behavior change here.

I agree it would be great if we'd have some other test case that wouldn't require those. I'll try to come up with some, it might be the easiest with `ObjectFlags.ContainsObjectOrArrayLiteral`.

Reply:
Hm, I have tried to come up with a test case that would rely on the change in how `ObjectFlags.ContainsObjectOrArrayLiteral` or `ObjectFlags.NonInferrableType` might get propagated but I failed.